Example/Tests/TensorIOModelBundleValidatorTests.mm

```diff
@@ -0,0 +1,2174 @@
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithoutOutputsDoesNotValidate {
+    // it should not validate
+    
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    [properties removeObjectForKey:@"outputs"];
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithValidPropertiesValidates {
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertTrue(valid);
+    XCTAssertNil(error);
+}
+
+// MARK: - Model Properties Validation
+
+- (void)testModelPropertiesValidates {
+    
+    NSDictionary *properties;
+    NSError *error;
+    BOOL valid;
+    
+    // it should not validate if model properties is empty
+    
+    properties = @{};
+    error = nil;
+    
+    valid = [self.modelValidator validateModelProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+    
+    // it should not validate if model properties has extraneous keys
+    
+    properties = @{
+        @"file": @"models.json",
+        @"quantized": @(YES),
+        @"foo": @""
+    };
+    error = nil;
+    
+    valid = [self.modelValidator validateModelProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+    
+    // it should not validate if file is not a string
+    
+    properties = @{
+        @"file": @(0),
+        @"quantized": @(YES),
+        @"foo": @""
+    };
+    error = nil;
+    
+    valid = [self.modelValidator validateModelProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+    
+    // it should not validate if quantized is not a bool
+    
+    properties = @{
+        @"file": @"model.json",
+        @"quantized": @"",
+    };
+    error = nil;
+    
+    valid = [self.modelValidator validateModelProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+    
+    // it should validate
+    
+    properties = @{
+        @"file": @"model.json",
+        @"quantized": @(YES)
+    };
+    error = nil;
+    
+    valid = [self.modelValidator validateModelProperties:properties error:&error];
+    
+    XCTAssertTrue(valid);
+    XCTAssertNil(error);
+    
+    // it should validate
+    
+    properties = @{
+        @"file": @"model.json",
+        @"quantized": @(YES),
+        @"type": @""
+    };
+    error = nil;
+    
+    valid = [self.modelValidator validateModelProperties:properties error:&error];
+    
+    XCTAssertTrue(valid);
+    XCTAssertNil(error);
+}
+
+// MARK: - Assets Validation
+
+- (void)testAnIncorrectlyNamedModelFileDoesNotValidate {
```

Comment:
Can we add model file validation to check that it is a valid TFLite file? This is not a blocker on this PR, and I'm not even sure if I would want it as a user (if, for example, it required us to run dummy data through a model). Just something to consider.

I don't know if there's an easy way (like `saved_model_cli` binary which comes with TensorFlow) to test TFLite models.

Reply:
Let's add this to 0.3.x. I'll make an issue for it. If nothing else we can try to load the model and if that fails report that to the user.